torchdynamo/symbolic_convert.py

```diff
@@ -146,7 +147,7 @@ def wrap_local(self, name, value):
                 state=TracingSupported.YES,
                 guards={Guard(name, GuardSource.LOCAL, GuardBuilder.VALUE_MATCH)},
             )
-        elif value is True or value is False or value is None:
+        elif value is True or value is False or value is None or isinstance(value, Real):
```

Comment:
This might not work exactly right.  There are two ways we could support int/floats:

1) Specialize the entire graph based on the exact value.  (The graph gets recompiled whenever the value changes.)  

This is what this PR does, however to make it work you will need to extend how `GuardBuilder.VALUE_MATCH`
works, you will notice it is actually checking `id(val1) == id(val2)`.   Which works for True/False/None because they are singletons, but will have false negatives for integers.  (`id` looks at the address of the object in memory.)  Adding this check should be an easy fix.

2) Treat the value as an input to the graph similar to how `TensorVariable` is implemented.  (See the handling of `isinstance(..., torch.Tensor)` above.)

For this we may want to set `TracingSupported.UNKOWN` (to prevent extraction of a graph with no Tensors) and set the type annotation correctly in the fx Graph.

-----

For None/True/False, going with option 1 makes sense -- as it will only specialize a few times.

For floats, option 2 might be better to prevent over-specialization. 

Integers could go either way.  I might lean towards option 2.  We could also make it configurable, or do some hybrid of specializing small integers but not big integers.